tracing/flags.go

```diff
@@ -0,0 +1,110 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/els0r/telemetry/tracing/internal/flagutil"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+const (
+	otelGRPC = "otel-grpc"
+	stdout   = "stdout"
+)
+
+var supportedCollectors = []string{
+	otelGRPC,
+	stdout,
+}
+
+// Tracing flag names and their help text. The '.' is the hierarchy delimiter
+const (
+	TracingKey = "tracing"
+
+	TracingEnabledArg     string = TracingKey + ".enabled"
+	TracingEnabledDefault bool   = true
+	TracingEnabledHelp    string = "enable tracing"
+
+	TracingCollectorKey string = TracingKey + ".collector"
+
+	TracingCollectorTypeArg     string = TracingCollectorKey + ".type"
+	TracingCollectorTypeDefault string = ""
+	TracingCollectorTypeHelp    string = "tracing collector type"
+
+	TracingCollectorEndpointArg     string = TracingCollectorKey + ".endpoint"
+	TracingCollectorEndpointDefault string = ""
+	TracingCollectorEndpointHelp    string = "endpoint collecting traces"
+)
+
+type traceFlagsConfig struct {
+	enabled           bool
+	collectorType     string
+	collectorEndpoint string
+}
+
+func (t traceFlagsConfig) options(ctx context.Context) (opts []Option) {
+	if !(t.enabled) {
+		return opts
```

Comment:
Is that a feature of some interface? Why are the options returned at all if tracing isn't enabled? :thinking:

Reply:
Because we later check if there are more than 0 options.

Reply:
Ah, yeah, NVM then...